src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Expressions/StringExpression.cs

```diff
@@ -134,9 +141,9 @@ public override Expression ToLinqExpression(Expression propertyExpr)
                 case Condition.NotIsEmpty:
                     return Expression.NotEqual(propertyExpr, Expression.Constant(string.Empty));
                 case Condition.IsNull:
-                    return Expression.Equal(propertyExpr, Expression.Constant(null));
+                    return Expression.ReferenceEqual(propertyExpr, Expression.Constant(null));
```

Comment:
Have you tested this code? Not sure if calling Expression.Constant(null) returns the same actual reference every time. This is just an observation, I am not entirely sure about this one.